order.go

```diff
@@ -39,32 +39,125 @@ type OrderServiceOp struct {
 	client *Client
 }
 
+type orderStatus string
+
+// https://shopify.dev/docs/api/admin-rest/2023-07/resources/order#get-orders?status=any
+const (
+	//Show only open orders.
+	StatusOpen orderStatus = "open"
+
+	//Show only closed orders.
+	StatusClosed orderStatus = "closed"
+
+	//Show only cancelled orders.
+	StatusCancelled orderStatus = "cancelled"
+
+	//Show orders of any status, open, closed, cancellerd, or archived.
+	StatusAny orderStatus = "any"
+)
+
+type orderFulfillmentStatus string
+
+// https://shopify.dev/docs/api/admin-rest/2023-07/resources/order#get-orders?status=any
+const (
+	//Show orders that have been shipped.
+	FulfillmentStatusShipped orderFulfillmentStatus = "shipped"
+
+	//Show partially shipped orders.
+	FulfillmentStatusPartial orderFulfillmentStatus = "partial"
+
+	//Show orders that have not yet been shipped.
+	FulfillmentStatusUnshipped orderFulfillmentStatus = "unshipped"
+
+	//Show orders of any fulfillment status.
+	FulfillmentStatusAny orderFulfillmentStatus = "any"
+
+	//Returns orders with fulfillment_status of null or partial.
+	FulfillmentStatusUnfulfilled orderFulfillmentStatus = "unfulfilled"
+
+	//"fulfilled" used to be an acceptable value? Was it deprecated? It isn't noted
+	//in the Shopify docs at the provided URL, but it was used in tests and still
+	//seems to function.
+	FulfillmentStatusFulfilled orderFulfillmentStatus = "fulfilled"
+)
+
+type orderFinancialStatus string
+
+// https://shopify.dev/docs/api/admin-rest/2023-07/resources/order#get-orders?status=any
+const (
+	//Show only authorized orders.
+	FinancialStatusAuthorized orderFinancialStatus = "authorized"
+
+	//Show only pending orders.
+	FinancialStatusPending orderFinancialStatus = "pending"
+
+	//Show only paid orders.
+	FinancialStatusPaid orderFinancialStatus = "paid"
+
+	//Show only partially paid orders.
+	FinancialStatusPartiallyPaid orderFinancialStatus = "partially_paid"
+
+	//Show only refunded orders.
+	FinancialStatusRefunded orderFinancialStatus = "refunded"
+
+	//Show only voided orders.
+	FinancialStatusVoided orderFinancialStatus = "voided"
+
+	//Show only partially refunded orders.
+	FinancialStatusPartiallyRefunded orderFinancialStatus = "partially_refunded"
+
+	//Show orders of any financial status.
+	FinancialStatusAny orderFinancialStatus = "any"
+
+	//Show authorized and partially paid orders.
+	FinancialStatusUnpaid orderFinancialStatus = "unpaid"
+)
+
+type orderCancelReason string
+
+const (
+	//The customer canceled the order.
+	CancelReasonCustomer orderCancelReason = "customer"
```

Comment:
Same as above, let's make this `OrderCancelReason...` for the consts.

Reply:
Done